frontend/packages/ux-editor/src/components/config/Expressions/ExpressionContent/ExpressionEditMode/ExpressionEditMode.tsx

```diff
@@ -0,0 +1,165 @@
+  updateComplexExpressionOnExpression,
+  updateOperatorOnExpression,
+  updateSubExpressionOnExpression,
+} from '../../../../../utils/expressionsUtils';
+import { ComplexExpression } from '../ComplexExpression';
+import { SimpleExpression } from '../SimpleExpression';
+import { Button, Switch } from '@digdir/design-system-react';
+import { CheckmarkIcon, PlusCircleIcon, TrashIcon } from '@navikt/aksel-icons';
+import { Trans } from 'react-i18next';
+import classes from '../ExpressionContent.module.css';
+import { stringifyData } from '../../../../../utils/jsonUtils';
+import { useText } from '../../../../../hooks';
+
+export interface ExpressionEditModeProps {
+  expression: Expression;
+  componentName: string;
+  onSetEditMode: (editMode: boolean) => void;
+  onDeleteExpression: (expression: Expression) => void;
+  onDeleteSubExpression: (subExpression: SubExpression) => void;
+  onSaveExpressionAndSetCheckMark: (expression: Expression) => void;
+  onSetExpression: (expression: Expression) => void;
+}
+
+export const ExpressionEditMode = ({
+  expression,
+  componentName,
+  onSetEditMode,
+  onDeleteExpression,
+  onDeleteSubExpression,
+  onSaveExpressionAndSetCheckMark,
+  onSetExpression,
+}: ExpressionEditModeProps) => {
+  const [freeStyleEditing, setFreeStyleEditing] = useState<boolean>(!!expression.complexExpression);
+  const t = useText();
+
+  const addSubExpression = (expressionOperator: Operator) => {
+    const newExpression: Expression = addSubExpressionToExpression(expression, expressionOperator);
+    onSetExpression(newExpression);
+  };
+
+  const updateOperator = (expressionOperator: Operator) => {
+    const newExpression: Expression = updateOperatorOnExpression(expression, expressionOperator);
+    onSetExpression(newExpression);
+  };
+
+  const updateSubExpression = (index: number, subExpression: SubExpression) => {
+    const newExpression: Expression = updateSubExpressionOnExpression(
+      expression,
+      index,
+      subExpression,
+    );
+    onSetExpression(newExpression);
+  };
+
+  const updateComplexExpression = (newComplexExpression: any) => {
+    const newExpression: Expression = updateComplexExpressionOnExpression(
+      expression,
+      newComplexExpression,
+    );
+    onSetExpression(newExpression);
+  };
+
+  const handleToggleFreeStyleEditing = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFreeStyleEditing(event.target.checked);
+    if (event.target.checked) {
+      const stringRepresentationOfExpression = stringifyData(externalExpression);
+      updateComplexExpression(stringRepresentationOfExpression);
+    } else {
+      updateComplexExpression(undefined);
+    }
+  };
+
+  const allowToSaveExpression =
+    (expression.subExpressions?.filter((subExp) => !subExp.function)?.length === 0 &&
+      expression.subExpressions.length !== 0 &&
+      !!expression.property) ||
+    (complexExpressionIsSet(expression.complexExpression) && !!expression.property);
```

Comment:
Dette vill jeg hatt i en separat funksjon på utsiden, gjerne med en egen enhetstest. Kanskje også delt den litt opp for å gjøre det tydelig hva som er kriteriene for at et uttrykk skal kunne lagres.